src/modules/emote_autocomplete/components/AutocompletePopover.jsx

```diff
@@ -0,0 +1,63 @@
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import classNames from 'classnames';
+import Popover from 'rsuite/Popover';
+import mergeRefs from 'react-merge-refs';
+import styles from './AutocompletePopover.module.css';
+import Emotes from './Emotes.jsx';
+import repositionPopover from '../../../utils/popover.js';
+import useResize from '../../../common/hooks/Resize.jsx';
+
+const TOP_PADDING = 2;
+
+const EmoteMenuPopover = React.forwardRef(
+  (
+    {
+      triggerRef,
+      appendToChat,
+      className,
+      style,
+      boundingQuerySelector,
+      chatInputElement,
+      onComplete,
+      getChatInputPartialEmote,
+      ...props
+    },
+    ref
+  ) => {
+    const localRef = useRef(null);
+    const [popoverWidth, setPopoverWidth] = useState(null);
+
+    function reposition() {
+      repositionPopover(localRef, boundingQuerySelector, TOP_PADDING);
+    }
+
+    useEffect(() => reposition(), [localRef, style]);
+
+    const callback = useCallback(() => {
+      const {width} = chatInputElement.getBoundingClientRect();
+      setPopoverWidth(width);
+      reposition();
```

Comment:
is there any reason not to combine this callback with the above reposition function?

```
const reposition = useCallback(() => {
  const {width} = chatInputElement.getBoundingClientRect();
  setPopoverWidth(width);
  repositionPopover(localRef, boundingQuerySelector, TOP_PADDING);
}, [chatInputElement, setPopoverWidth, localRef, boundingQuerySelector]);

useResize(reposition);
useEffect(reposition, [reposition])
```